lib/stores/application_store.dart

```diff
@@ -404,25 +407,90 @@ abstract class _ApplicationStore with Store {
 
   @action
   Future<void> updateTransactions(List<TransactionDto> transactions) async {
-    for (var i = 0; i < transactions.length; i++) {
-      var currentIndex = currentTransactions
-          .indexWhere((element) => element.id == transactions[i].id);
-      if (currentIndex == -1) {
-        currentTransactions
-            .add(TransactionVm.fromTransactionDto(transactions[i]));
+    _addOrUpdateCurrentTransactions(transactions);
+    _addStoredTransactionsToCurrent();
+  }
+
+  void _addOrUpdateCurrentTransactions(List<TransactionDto> transactions) {
+    for (var transaction in transactions) {
+      var index = currentTransactions
+          .indexWhere((element) => element.id == transaction.id);
+      var transactionVm = TransactionVm.fromTransactionDto(transaction);
+      if (index == -1) {
+        currentTransactions.add(transactionVm);
       } else {
-        currentTransactions[currentIndex] =
-            TransactionVm.fromTransactionDto(transactions[i]);
+        currentTransactions[index] = transactionVm;
       }
     }
   }
 
+  @action
+  void _addStoredTransactionsToCurrent() {
+    // Add transactions that are not in currentTransactions in order
+    for (var trx in _hiveStorage.storedTransactions.values) {
+      if (!currentTransactions.any((element) => element.id == trx.id)) {
+        int insertIndex = currentTransactions.indexWhere(
+          (element) => element.targetTick > trx.targetTick,
+        );
+        if (insertIndex == -1) {
+          currentTransactions.add(trx);
+        } else {
+          currentTransactions.insert(insertIndex, trx);
+        }
+      }
+    }
+  }
+
+  @action
+  addStoredTransaction(TransactionVm transaction) {
+    _hiveStorage.addStoredTransaction(transaction);
+  }
+
+  @action
+  void validatePendingTransactions(int currentTick) {
+    List<TransactionVm> toBeRemoved = [];
+    for (var trx in _hiveStorage.storedTransactions.values) {
+      if (currentTick >= trx.targetTick + 5) {
+        /// If any pending transaction is older than the current tick, convert it
+        /// to invalid (ignored by network)
+        if (trx.isPending) {
+          convertPendingToInvalid(trx);
+        }
+
+        /// If the stored transaction is in currentTransactions and successful
+        /// then remove it from local storage
```

Comment:
@ahmed-tarek-salem  if it's returned as failed from from the network, it should also be deleeted from the local storage.

Reply:
The application calculates the status as in the image, which means even if it's failed it returns as "Success"

<img width="486" alt="image" src="https://github.com/user-attachments/assets/96b730b8-220c-48af-95c5-24e6273b04f9" />